pkg/scheduler/framework/plugins/nodevolumelimits/csi.go

```diff
@@ -81,10 +82,29 @@ func (pl *CSILimits) EventsToRegister() []framework.ClusterEventWithHint {
 		// because any new CSINode could make pods that were rejected by CSI volumes schedulable.
 		{Event: framework.ClusterEvent{Resource: framework.CSINode, ActionType: framework.Add}},
 		{Event: framework.ClusterEvent{Resource: framework.Pod, ActionType: framework.Delete}},
-		{Event: framework.ClusterEvent{Resource: framework.PersistentVolumeClaim, ActionType: framework.Add}},
+		{Event: framework.ClusterEvent{Resource: framework.PersistentVolumeClaim, ActionType: framework.Add}, QueueingHintFn: pl.isSchedulableAfterPVCAdded},
 	}
 }
 
+func (pl *CSILimits) isSchedulableAfterPVCAdded(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	_, addedPvc, err := util.As[*v1.PersistentVolumeClaim](oldObj, newObj)
+	if err != nil {
+		return framework.Queue, fmt.Errorf("unexpected objects in isSchedulableAfterPVCAdded: %w", err)
+	}
+
+	if addedPvc.Namespace != pod.Namespace {
+		return framework.QueueSkip, nil
+	}
+
+	for _, volumes := range pod.Spec.Volumes {
```

Comment:
Please consider `pod.spec.volumes.ephemeral.volumeClaimTemplate`. A helper function can be found in https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/component-helpers/storage/ephemeral/ephemeral.go#L49

Reply:
@carlory `VolumeIsForPod` requires `*v1.PersistentVolumeClaim` but, we can't get it since the PVCs that `deletedPod` has had been already deleted.

Reply:
@utam0k okay.

Reply:
For example, 

```go
switch {
case volume.PersistentVolumeClaim != nil:
	pvcName = vol.PersistentVolumeClaim.ClaimName
case volume.Ephemeral != nil:
	pvcName = ephemeral.VolumeClaimName(pod, &volume)
default:
	// Volume is not using a PVC, ignore
	continue
}

if pvcName == newPVC.Name {
	return framework.Queue, nil
}
```

Reply:
+1
https://github.com/kubernetes/kubernetes/pull/124703/commits/3347883ddc0cf561e35c3e3adc6a62d898e84347